src/wire/arp.rs

```diff
@@ -237,101 +236,233 @@ impl<T: AsRef<[u8]> + AsMut<[u8]>> Packet<T> {
         match self {
-            &Repr::EthernetIpv4 {
-                operation,
-                source_hardware_addr, source_protocol_addr,
-                target_hardware_addr, target_protocol_addr
+            &Repr::EthernetIpv4Request {
+                source_hardware_addr,
+                source_protocol_addr,
+                target_hardware_addr,
+                target_protocol_addr,
             } => {
                 packet.set_hardware_type(Hardware::Ethernet);
                 packet.set_protocol_type(Protocol::Ipv4);
                 packet.set_hardware_len(6);
                 packet.set_protocol_len(4);
-                packet.set_operation(operation);
+                packet.set_operation(Operation::Request);
                 packet.set_source_hardware_addr(source_hardware_addr.as_bytes());
                 packet.set_source_protocol_addr(source_protocol_addr.as_bytes());
                 packet.set_target_hardware_addr(target_hardware_addr.as_bytes());
                 packet.set_target_protocol_addr(target_protocol_addr.as_bytes());
-            },
-            &Repr::__Nonexhaustive => unreachable!()
+            }
+            &Repr::EthernetIpv4Reply {
+                source_hardware_addr,
+                source_protocol_addr,
+                target_hardware_addr,
+                target_protocol_addr,
+            } => {
+                packet.set_hardware_type(Hardware::Ethernet);
+                packet.set_protocol_type(Protocol::Ipv4);
+                packet.set_hardware_len(6);
+                packet.set_protocol_len(4);
+                packet.set_operation(Operation::Reply);
+                packet.set_source_hardware_addr(source_hardware_addr.as_bytes());
+                packet.set_source_protocol_addr(source_protocol_addr.as_bytes());
+                packet.set_target_hardware_addr(target_hardware_addr.as_bytes());
+                packet.set_target_protocol_addr(target_protocol_addr.as_bytes());
+            }
+            &Repr::__Nonexhaustive => unreachable!(),
         }
     }
 }
 
-impl<T: AsRef<[u8]>> fmt::Display for Packet<T> {
+impl<T: Payload> fmt::Display for Packet<T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match Repr::parse(self) {
             Ok(repr) => write!(f, "{}", repr),
             _ => {
                 write!(f, "ARP (unrecognized)")?;
-                write!(f, " htype={:?} ptype={:?} hlen={:?} plen={:?} op={:?}",
-                       self.hardware_type(), self.protocol_type(),
-                       self.hardware_len(), self.protocol_len(),
-                       self.operation())?;
-                write!(f, " sha={:?} spa={:?} tha={:?} tpa={:?}",
-                       self.source_hardware_addr(), self.source_protocol_addr(),
-                       self.target_hardware_addr(), self.target_protocol_addr())?;
+                write!(
+                    f,
+                    " htype={:?} ptype={:?} hlen={:?} plen={:?} op={:?}",
```

Comment:
Die (ich nehme an Auto-)Formattierung ist irgendwie komisch. Aber noch in Ordnung.